### PR TITLE
update doc for insert select pushdown to tiflash

### PR DESCRIPTION
**Pull request by gengliqi**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1687,6 +1683,7 @@ Query OK, 0 rows affected (0.09 sec)
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
```
Comment (qiancai) [inline system-variables.md:1685]: L1681 的默认值需要改成 ON

File: system-variables.md [hunk 2]
```diff
@@ -1752,6 +1749,7 @@ Query OK, 0 rows affected (0.09 sec)
 - 这个变量用于控制是否忽略优化器代价估算，强制使用 TiFlash 的 MPP 模式执行查询，可以设置的值包括：
     - 0 或 OFF，代表不强制使用 MPP 模式（默认）
     - 1 或 ON，代表将忽略代价估算，强制使用 MPP 模式。注意：只有当 `tidb_allow_mpp=true` 时该设置才生效。
+- 从 v7.1.0 开始，TiDB 要求只有当前会话的 SQL Mode 不包含 `STRICT_TRANS_TABLES`, `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的查询下推至 TiFlash。开启 `tidb_enforce_mpp` 会忽略该情况下 `STRICT SQL Mode` 的限制。
```
Comment (qiancai) [inline system-variables.md:1752]: “开启 `tidb_enforce_mpp` 会忽略该情况下 `STRICT SQL Mode` 的限制。” 是指，只要开启 `tidb_enforce_mpp`，即使在 `STRICT SQL Mode` 下，`INSERT INTO SELECT` 语句中的 `SELECT` 子句也会吓推到 TiFlash 吗？

Comment (gengliqi) [reply system-variables.md]: 这个删了，之前误解了 pm 的意思。就算开启了 `tidb_enforce_mpp`，也是需要满足不是 `STRICT SQL Mode` 才能下推的。

File: system-variables.md [hunk 3]
```diff
@@ -1675,18 +1675,15 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用于控制包含增删改的 SQL 语句中的读取操作能否下推到 TiFlash，比如：
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
+- 从 v7.1.0 开始，该变量废弃，其取值将固定为默认值 ON，即默认开启 `tidb_enable_tiflash_read_for_write_stmt`。另外 TiDB 要求只有当前会话的 SQL Mode 不包含 `STRICT_TRANS_TABLES`, `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的查询下推至 TiFlash。
```
Comment (qiancai) [inline system-variables.md:1686]:
```suggestion
- 从 v7.1.0 开始，该变量废弃。当 TiDB 执行 `INSERT INTO SELECT` 语句中的 `SELECT` 子句时，优化器将根据 SQL Mode 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。 需要注意的是，TiDB 要求只有当前会话的 [SQL 模式](/sql-mode.md) 不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的查询下推至 TiFlash。
```

File: tiflash/tiflash-results-materialization.md [hunk 4]
```diff
@@ -15,8 +15,7 @@ summary: 介绍如何在同一个事务中保存 TiFlash 的查询结果。
 
 > **注意：**
 >
-> - 默认情况下 ([`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入))，TiDB 优化器将依据查询代价智能选择下推查询到 TiKV 或 TiFlash。如需强制使用 TiFlash 查询，你可以设置系统变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入) 为 `ON`。
-> - 在实验特性阶段，该功能默认关闭。要开启此功能，请设置系统变量 [`tidb_enable_tiflash_read_for_write_stmt`](/system-variables.md#tidb_enable_tiflash_read_for_write_stmt-从-v630-版本开始引入) 为 `ON`。
+> - TiDB 要求只有当前会话的 SQL Mode 不包含 `STRICT_TRANS_TABLES`, `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，才允许将 `INSERT INTO SELECT` 语句中的查询下推至 TiFlash。满足 SQL Mode 要求后，TiDB 优化器将依据查询代价智能选择是否下推到 TiFlash。如需忽略优化器代价估算强制使用 TiFlash 查询，你可以设置系统变量 [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入) 为 `ON`。注意：只有当 `tidb_allow_mpp=true` 时该设置才生效。
```
Comment (qiancai) [inline tiflash/tiflash-results-materialization.md:18]: 这里也需要对应改下 @gengliqi

File: system-variables.md [hunk 5]
```diff
@@ -1675,18 +1675,15 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用于控制包含增删改的 SQL 语句中的读取操作能否下推到 TiFlash，比如：
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
+- 从 v7.1.0 开始，该变量废弃，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，TiDB 要求只有当前会话的 [SQL 模式](/sql-mode.md) 不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的读取操作下推至 TiFlash。
```
Comment (qiancai) [inline system-variables.md:1686]:
```suggestion
- 从 v7.1.0 开始，该变量废弃，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，TiDB 才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的读取操作下推至 TiFlash。
```

File: system-variables.md [hunk 6]
```diff
@@ -1675,18 +1675,15 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用于控制包含增删改的 SQL 语句中的读取操作能否下推到 TiFlash，比如：
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
+- 从 v7.1.0 开始，该变量废弃，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，TiDB 才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的读取操作下推至 TiFlash。
```
Comment (qiancai) [inline system-variables.md:1686]:
```suggestion
- 从 v7.1.0 开始，该变量废弃。当 [`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 时，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`，即不是 `STRICT SQL Mode` 时，TiDB 才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的读取操作下推至 TiFlash。
```

File: system-variables.md [hunk 7]
```diff
@@ -1675,18 +1675,15 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用于控制包含增删改的 SQL 语句中的读取操作能否下推到 TiFlash，比如：
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
+- 从 v7.1.0 开始，该变量废弃。当 [`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 时，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 为非严格模式（即 `sql_mode` 值不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`），TiDB 才允许将包含增删改的 SQL 语句(如 `INSERT INTO SELECT`)中的读取操作下推至 TiFlash。
```
Comment (ran-huang) [inline system-variables.md:1686]:
```suggestion
- 从 v7.1.0 开始，该变量废弃。当 [`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 时，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 为非严格模式（即 `sql_mode` 值不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`）时，TiDB 才允许将包含增删改的 SQL 语句（如 `INSERT INTO SELECT`）中的读取操作下推至 TiFlash。
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 8]
```diff
@@ -1675,18 +1675,15 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
 - 这个变量用于控制包含增删改的 SQL 语句中的读取操作能否下推到 TiFlash，比如：
 
     - `INSERT INTO SELECT` 语句中的 `SELECT` 查询（典型应用场景为 [TiFlash 查询结果物化](/tiflash/tiflash-results-materialization.md)）
     - `UPDATE` 和 `DELETE` 语句中的 `WHERE` 条件过滤
+- 从 v7.1.0 开始，该变量废弃。当 [`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 时，优化器将根据 [SQL 模式](/sql-mode.md) 及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md) 为非严格模式（即 `sql_mode` 值不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`）时，TiDB 才允许将包含增删改的 SQL 语句（如 `INSERT INTO SELECT`）中的读取操作下推至 TiFlash。
```
Comment (Oreoxmt) [inline system-variables.md:1686]:
```suggestion
- 从 v7.1.0 开始，该变量废弃。当 [`tidb_allow_mpp = ON`](/system-variables.md#tidb_allow_mpp-从-v50-版本开始引入) 时，优化器将根据 [SQL 模式](/sql-mode.md)及 TiFlash 副本的代价估算自行决定是否将查询下推至 TiFlash。需要注意的是，只有当前会话的 [SQL 模式](/sql-mode.md)为非严格模式（即 `sql_mode` 值不包含 `STRICT_TRANS_TABLES` 和 `STRICT_ALL_TABLES`）时，TiDB 才允许将包含增删改的 SQL 语句（如 `INSERT INTO SELECT`）中的读取操作下推至 TiFlash。
```

**Pull request merged**